Rakefile

```diff
@@ -86,3 +86,211 @@ EOM
+    write_to_inventory_file(inventory_hash, 'inventory.yaml')
+  end
+
+  desc 'Installs the module on the master'
+  task :install_module do
+    on_master { |master| Rake::Task['litmus:install_module'].invoke(master.uri) }
+  end
+
+  desc 'Set up the test infrastructure'
+  task :setup do
+    tasks = [
+      :provision_vms,
+      :setup_pe,
+      :setup_servicenow_instance,
+      :setup_servicenow_cmdb,
+      :install_module,
+    ]
+
+    tasks.each do |task|
+      task = "acceptance:#{task}"
+      puts("Invoking #{task}")
+      Rake::Task[task].invoke
+      puts("")
+    end
+  end
+
+  desc 'Runs the tests'
+  task :run_tests do
+    puts("Running the tests ...\n")
+
+    # TODO: We can't use litmus:acceptance:parallel here b/c that will try
+    # to run the spec tests on _all_ targets in the inventory.yaml file,
+    # including our ServiceNow instance. Until litmus:acceptance:parallel
+    # has a way for you to pass-in specific targets to run the tests on,
+    # we'll just run these tests via rspec
+    system({ 'TARGET_HOST' => on_master(&:uri) }, 'bundle exec rspec ./spec/acceptance')
+  end
+
+  desc 'Teardown the setup'
+  task :tear_down do
+    puts("Tearing down the test infrastructure ...\n")
+
+    inventory_hash = LitmusHelpers.inventory_hash_from_inventory_file
+
+    # First, delete all created CMDB entries from the given ServiceNow
+    # instance
+    begin
+      on_servicenow_instance do |instance|
+        # Skip if the instance URI points to the mock ServiceNow instance
+        # since that will be destroyed when we teardown the master via
+        # Litmus
+        next if instance.uri =~ Regexp.new(Regexp.escape(on_master(&:uri)))
+        puts("Deleting test-specific CMDB records from #{instance.uri} ...")
+
+        record_ids = cmdb_record_ids
+        record_ids.each do |uri, sys_id|
+          begin
+            puts("Deleting #{uri}'s CMDB record (sys_id = #{sys_id}) ...")
+            instance.run_bolt_task(
+              'servicenow_tasks::delete_record',
+              { 'table' => 'cmdb_ci', 'sys_id' => sys_id },
+            )
+            record_ids.delete(sys_id)
+          rescue => e
+            # Update the inventory.yaml file to remove deleted CMDB records
+            inventory_hash['vars']['cmdb_record_ids'] = record_ids
+            write_to_inventory_file(inventory_hash, 'inventory.yaml')
+            raise e
+          end
+        end
+      end
+    rescue TargetNotFoundError
+      # Pass-thru, this means that the tear_down task was called before
+      # the ServiceNow instance was setup
+    end
+
+    # Now teardown the master
+    Rake::Task['litmus:tear_down'].invoke(on_master(&:uri))
+
+    # Delete the inventory file
+    FileUtils.rm_f('inventory.yaml')
+  end
+
+  desc 'Task for CI'
+  task :ci_run_tests do
+    begin
+      Rake::Task['acceptance:setup'].invoke
+      Rake::Task['acceptance:run_tests'].invoke 
```

Comment:
Unfortunately I deleted my `inventory.yaml` file and now vmpooler's failing to get the right Ubuntu VM (on the API end, not us) so I'll post a high-level 'final' `inventory.yaml` after setup runs:
```
- groups:
      - name: ssh_nodes
         targets:
           - <master_vm_bolt_config>
      - name: servicenow_nodes
         targets:
           - <servicenow_instance_bolt_config> # Based on v2 inventory in https://github.com/puppetlabs/servicenow_tasks#setup
vars:
  cmdb_record_ids:
    <master_uri>: <master_cmdb_record_id>
```

I'll update this once vmpooler's working again.

Reply:
Oops, I accidentally resolved a conversation I wasn't involved in. Sorry.